app/models/manageiq/providers/nuage/network_manager/cloud_subnet.rb

```diff
@@ -1,2 +1,60 @@
 class ManageIQ::Providers::Nuage::NetworkManager::CloudSubnet < ::CloudSubnet
+  supports :delete
+  supports :create
+
+  def delete_cloud_subnet_queue(userid)
+    task_opts = {
+      :action => "deleting Cloud Subnet for user #{userid}",
+      :userid => userid
+    }
+    queue_opts = {
+      :class_name  => self.class.name,
+      :method_name => 'raw_delete_cloud_subnet',
+      :instance_id => id,
+      :priority    => MiqQueue::HIGH_PRIORITY,
+      :role        => 'ems_operations',
+      :zone        => ext_management_system.my_zone,
+      :args        => []
+    }
+    MiqTask.generic_action_with_callback(task_opts, queue_opts)
+  end
+
+  def raw_delete_cloud_subnet
+    $nuage_log.info("Deleting Cloud Subnet (ems_ref = #{ems_ref})")
+
+    Ansible::Runner.run(
+      ext_management_system.ansible_env_vars,
+      ext_management_system.ansible_extra_vars(:id => ems_ref),
+      ext_management_system.playbook('remove-subnet.yml')
+    )
+
+    $nuage_log.info("Done deleting Cloud Subnet (ems_ref = #{ems_ref})")
+  rescue => e
+    $nuage_log.error "Error deleting Cloud Subnet: #{e}"
+    raise MiqException::MiqCloudSubnetDeleteError
+  end
+
+  def self.raw_create_cloud_subnet(ext_management_system, options)
+    $nuage_log.info("Create Cloud Subnet (ems_ref = #{ems_ref})")
+
+    # TODO: update UI to match what Nuage needs
```

Comment:
@Ladas @agrare I remember you mentioned that UI work is still in progress for AnsibleRunner - does this mean there will be different UI approach for AnsibleRunner inputs? Or should I just go and update the current UI as usually with a few more `ng-if`s to match inputs for creating new Nuage CloudSubnet?

/cc @himdel

Reply:
If there is an existing button and action for these methods I don't think you need to make any UI changes

Reply:
The new way, that the UI team prepares, is that you can have your own javascript code, with the specific form, under the Nuage provider codebase

Reply:
@martinpovolny looks like this will also touch list toolbars, we can't do the provider specific forms there, right?

Reply:
@miha-plesko : We are indeed working on a way to enable the provider authors such as you to independently extend ManageIQ with Ansible. From the UI perspective it right now means enabling providers to add new button groups to toolbars and specificly toolbars displayed in details screens of their relevant entities. In your case it might be a nuage network. When such a button is pressed there are possibilities what will happen.
 1. an API call will be fired (backed by some Ansible playbook)
 2. a dialog will be opened that will allow the user to enter additional information needed for an API call (backed by some Ansible playbook)

For the 2. there are 2 options:
 a) provider team will implement their own dialog in the form of a React class 
 b) provider team will provide a dynamic dialog definition (dynamic dialogs = service dialogs) and the UI will "play" the dynamic dialog.

In case a) the React class is responsible for executing the API call
In case b) data from the dynamic dialog run will be passed to an API endpoint


Some part of that can be seen with some example here:
https://github.com/ManageIQ/manageiq-ui-classic/pull/4262
and here
https://github.com/ManageIQ/manageiq-ui-classic/pull/4231


This should be a way the provider teams can implement (non-core most likely) functionality (mostly) on their own from their repos.

Having said that I don't think that it's what you should do for network subnet. But if you want to go that way we will try to help you.

Reply:
Regarding ng-ifs and provider name: We don't want to get to a spaghetti state (that we already have in places). Specifically I am afraid of provider name tests spread between Ruby and Javascript code. That get's quickly unmaintainable.

Ideally providers should have capabilities and the UI should query those capabilities and behave according to such capabilities and not based on matching a substring in provider name for example.

I perfectly understand that in some cases this is not possible and there needs to be a test for provider name somewhere. I'd just prefer that to be in as few places as possible.

For example if a form needs to change based on provider type (and there are no provider capabilitied that can be tested) you can create a form configuration hash that will drive the indivitual UI elements and have that hash pre-filled bases on provider type on the Ruby side in controller rather than create complex conditions on the javascript side with many ifs and ands. The javascript side then would not contain knowledge about the provider but would instead offer flexibility needed to satisfy different providers' needs.

It works better to talk code with examples rather than talk words in this case so I suggest you ping me early when you start extending existing form so that we can discuss the possibilities w/o you spending time and being afraid that your work would not get merged.